include/engine/api/nearest_result.hpp

```diff
@@ -0,0 +1,24 @@
+#ifndef OSRM_ENGINE_API_NEAREST_RESULT_HPP
+#define OSRM_ENGINE_API_NEAREST_RESULT_HPP
+
+#include "engine/api/waypoint.hpp"
+
+#include <vector>
+
+namespace osrm
+{
+namespace engine
+{
+namespace api
+{
+
+struct NearestResult
+{
+    std::vector<Waypoint> waypoints;
+};
```

Comment:
Because this will become the new interface for libosrm should we (this time) abstract away impl. details behind a proper class we can easily extend down the line? (Not specifically for the Nearest plugin more a general comment)